### ISSUE TITLE
False positive for E1136 (unsubscriptable-object) when using NewType to wrap a subscriptable collection

### ISSUE THREAD

**Issue by phelps-sg** [pylint-dev/pylint#8688]
### Bug description

In the code below, pylint reports ` E1136: Value 'f()' is unsubscriptable (unsubscriptable-object)`. However, the code runs and the assertion passes.

~~~python
from typing import NewType, Dict

TestType = NewType("TestType", Dict[str, int])


def f() -> TestType:
    return TestType({"a": 0})


assert f()["a"] == 0

~~~

### Command used

```shell
`pylint pylint-test.py`
```

### Pylint output

```shell
************* Module pylint-test
pylint-test.py:10:7: E1136: Value 'f()' is unsubscriptable (unsubscriptable-object)

-----------------------------------
Your code has been rated at 0.00/10
```

### Expected behavior

The code should pass the pylint inspections

### Pylint version

```shell
~~~
pylint 2.17.4
astroid 2.15.4
Python 3.10.11 | packaged by conda-forge | (main, May 10 2023, 18:58:44) [GCC 11.3.0]
~~~
```

**Comment by jacobtylerwalls** [comment]
Duplicate of #4944
